src/Resources/ResourceManager/Implementation/Deployments/NewAzureDeploymentWhatIfCmdlet.cs

```diff
@@ -0,0 +1,125 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation.Deployments
+{
+    using System;
+    using System.Management.Automation;
+    using Common.ArgumentCompleters;
+    using Management.ResourceManager.Models;
+    using SdkModels.Deployments;
+    using WindowsAzure.Commands.Utilities.Common;
+
+    /// <summary>
+    /// Creates a new deployment what-if.
+    /// </summary>
+    [Cmdlet(VerbsCommon.New,
+         ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "DeploymentWhatIf", SupportsShouldProcess = true,
+         DefaultParameterSetName = SubscriptionParameterSetWithParameterlessTemplateFile),
+     OutputType(typeof(PSWhatIfOperationResult))]
+    public class NewAzureDeploymentWhatIfCmdlet : DeploymentWhatIfCmdletWithParameters, IDynamicParameters
+    {
+        [Alias("DeploymentName")]
+        [Parameter(Mandatory = false,
+            HelpMessage = "The name of the deployment it's going to create. Only valid when a template is used. When a template is used, if the user doesn't specify a deployment name, use the current time, like \"20131223140835\".")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = false, HelpMessage = "The deployment scope type.")]
```

Comment:
These were just copied from the `New-AzDeployment` cmdlet. I'll cleanup the `ValueFromPipelineByPropertyName = false`. However, for the rest of them I don't want to any assumption, since the cmldet applies to the same use cases of `New-AzDeployment` cmdlet. Maybe @Tiano2017 has a better answer?

Reply:
@Tiano2017 could you answer the question here: whether this cmdlet has any piping scenarios?

Reply:
Had a offline discussion with @Tiano2017 and he said we need to preserve them. One possible scenario is that a user runs a Get-AzDeployment first, changes some properties (such as Location), and pipes it to the what-if cmdlet.